client/engine/engine.go

```diff
@@ -331,7 +339,7 @@ func (e *Engine) handleAPIEvent(apiEvent APIEvent) (ObjectiveChangeEvent, error)
 			e.metrics.RecordObjectiveStarted(request.Id(*e.store.GetAddress()))
 			ddfo, err := directdefund.NewObjective(request, true, e.store.GetConsensusChannelById)
 			if err != nil {
-				return ObjectiveChangeEvent{}, fmt.Errorf("handleAPIEvent: Could not create objective for %+v: %w", request, err)
+				return ObjectiveChangeEvent{ErredObjectives: []protocols.ObjectiveId{request.Id(e.metrics.me)}}, fmt.Errorf("handleAPIEvent: Could not create objective for %+v: %w", request, err)
```

Comment:
```suggestion
				return ObjectiveChangeEvent{ErredObjectives: []protocols.ObjectiveId{request.Id(*e.store.GetAddress())}}, fmt.Errorf("handleAPIEvent: Could not create objective for %+v: %w", request, err)
```

1. Is this equivalent to what you have?
2. Is it better this way, to not mix production and metrics code?

Reply:
1. in fact, anything you pass into this function would be equivalent - the argument isn't used by the function! (todo)
2. the `metrics` property was new to me, but it seems generally hard-wired into production methods. See `engine.Run`

In any case, your suggestion is an improvement.